tests/arch/arm/arm_interrupt/src/arm_interrupt.c

```diff
@@ -191,6 +191,13 @@ void arm_isr_handler(const void *args)
 {
 	ARG_UNUSED(args);
 
+#if defined(CPU_HAS_FPU) && defined(ARM)
```

Comment:
When wouldn't `ARM` be defined for this specific code/test?

Reply:
Agreed, the directive here should be: 
`#if defined(CONFIG_CPU_CORTEX_M) && defined(CONFIG_FPU) && defined(CONFIG_FPU_SHARING)`

CPU_HAS_FPU is not good here, cause the SoC may support FPU, but if the build doesn't, there is no need to clear any FP system control registers, and in fact, it may lead to co-processor faults.

The test is currently available for Cortex-M, so the CONFIG_CPU_CORTEX_M is currently redundant, but eventually we may want to support Cortex-R and 32 bit Cortex-A, hence my suggestion to add it.

Reply:
done